### ISSUE TITLE
Function supplied to `defaultItem` is not executed (using inline-blocks-demo)

### ISSUE THREAD

**Issue by sneyed** [tinacms/tinacms#1640]
## Description

[The docs](https://tina.io/docs/ui/inline-editing/inline-blocks/#part-2-block-template) say that the `defaultItem` key on the `BlockTemplate` interface can be either an object or a function that returns an object. 

![image](https://user-images.githubusercontent.com/14341953/100199629-2ed8a480-2f49-11eb-8327-39f0a94e9609.png)

However, when trying to supply a function to `defaultItem`, the function is not executed and the created block is not populated with any data.

### Steps to reproduce

1. Clone and checkout the `final-state` branch of the `inline-blocks-demo`
```bash
git clone https://github.com/tinacms/inline-blocks-demo.git
cd inline-blocks-demo
git checkout final-state
yarn install
yarn dev
```

2. Open `inline-blocks-demo\src\components\FeatureList.js` and modify `defaultItem` to be a function
```javascript
function FeatureList({ index }) { ... }

const FEATURE_BLOCKS = { ... };

export const featureListBlock = {
  ...
  template: {
    ...
    defaultItem: () => {
      console.log("This should print to console")
      return { ... }
    },
    fields: [],
  },
};
```

3. Save the file
4. Add a new feature list block at the bottom of the page

### Expected result

- The feature list is created and populated with default values 
- "This should print to console" appears in developer console

### Actual result

- The feature list is created but is not populated with any values
- Nothing is printed to developer console

#### Tinacms Package Versions
See package.json of https://github.com/tinacms/inline-blocks-demo

#### Tools

| Tool    | Version |
| ------- | ------- |
| browser |   Chrome      |
| os      |  Windows       |
| node    |   12.18.2      |
| npm     |   6.14.5      |

**Comment by sneyed** [comment]
I'm not 100% confident, but i suspect this issue may be related to this area of `react-tinacms-inline` ?

https://github.com/tinacms/tinacms/blob/c9cf9980034d7edb49b6afe8ff6b0fe9edf93c55/packages/react-tinacms-inline/src/blocks/add-block-menu.tsx#L68

**Comment by spbyrne** [comment]
Your hunch is correct. It seems the regular block field uses this code to run the defaultItem function if present:

```js
/* BlockFieldPlugin.tsx */

if (typeof template.defaultItem === 'function') {
        obj = template.defaultItem()
} else {
        obj = template.defaultItem || {}
}
```

But the inline block field doesn't use this logic. Since we have the proper code for the regular blocks field it shouldn't be hard to fix this.

**Comment by sneyed** [comment]
Note for future readers:
This can be closed when #1649 is merged
